### PR TITLE
Android: Build with vulkan source files

### PR DESCRIPTION
**Pull request by gaochun**
Build with vulkan backend source files and add corlor format mapping from gbm to vulkan

### CODE REVIEW

File: os/android/platformdefines.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "platformdefines.h"
+
+#ifdef USE_VK
+#include <drm/drm_fourcc.h>
+#endif
```
Comment (strassek) [inline os/android/platformdefines.cpp]: Why close this ifdef just to open it 2 lines later?

File: os/linux/platformdefines.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "platformdefines.h"
+
+#ifdef USE_VK
+#include <gbm.h>
+#endif
```
Comment (strassek) [inline os/linux/platformdefines.cpp]: Why close this ifdef just to open it 2 lines later?

File: os/platformcommondefines.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#ifndef OS_LINUX_PLATFORMCOMMONDEFINES_H_
+#define OS_LINUX_PLATFORMCOMMONDEFINES_H_
+
+#ifdef USE_VK
+#include <vulkan/vulkan.h>
+#endif
```
Comment (strassek) [inline os/platformcommondefines.h]: Why close this ifdef just to open it 2 lines later?

File: os/platformcommondefines.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+#ifdef USE_VK
+VkFormat NativeToVkFormat(int drm_format);
+#endif
+
+
+#endif  // OS_LINUX_PLATFORMCOMMONDEFINES_H_
```
Comment (strassek) [inline os/platformcommondefines.h]: I think we only use 1 newline before closing the ifdef/namespace

File: os/android/platformdefines.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    case DRM_FORMAT_ABGR8888:
+        return VK_FORMAT_R8G8B8A8_UNORM;
+    case DRM_FORMAT_XRGB2101010:
+    case DRM_FORMAT_ARGB2101010:
+        return VK_FORMAT_A2R10G10B10_UNORM_PACK32;
+    case DRM_FORMAT_XBGR2101010:
+    case DRM_FORMAT_ABGR2101010:
+        return VK_FORMAT_A2B10G10R10_UNORM_PACK32;
+    default:
+      ETRACE("%s: drm_format %d unhandled\n", __func__, drm_format);
```
Comment (strassek) [inline os/android/platformdefines.cpp]: You can drop the __func__, ETRACE already includes it

File: os/linux/platformdefines.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    case GBM_FORMAT_ABGR8888:
+        return VK_FORMAT_R8G8B8A8_UNORM;
+    case GBM_FORMAT_XRGB2101010:
+    case GBM_FORMAT_ARGB2101010:
+        return VK_FORMAT_A2R10G10B10_UNORM_PACK32;
+    case GBM_FORMAT_XBGR2101010:
+    case GBM_FORMAT_ABGR2101010:
+        return VK_FORMAT_A2B10G10R10_UNORM_PACK32;
+    default:
+      ETRACE("%s: gbm_format %d unhandled\n", __func__, gbm_format);
```
Comment (strassek) [inline os/linux/platformdefines.cpp]: You can drop the __func__, ETRACE already includes it

File: os/android/platformdefines.h [hunk 7]
```diff
@@ -32,6 +32,10 @@
 
 #include <ui/GraphicBuffer.h>
 
+#ifdef USE_VK
+#include <gbm.h>
```
Comment (kalyankondapally) [inline os/android/platformdefines.h]: You dont need this, we dont use gbm on Android.

File: os/android/platformdefines.cpp (hunk 1 above)
Comment (gaochun) [inline os/android/platformdefines.cpp]: I think there would be more include instructions and function definitions in this file when necessary, so uses separated ifdef for include and function.

**Comment by gaochun** [comment]
Rebased and fixed according to the comments.

File: os/android/platformdefines.h (hunk 7 above)
Comment (kalyankondapally) [inline os/android/platformdefines.h]: @gaochun You might have missed this.

**Comment by kalyankondapally** [comment]
@gaochun Can you rebase this and fix the comment. Would like to enable Vulkan as soon as possible.

File: os/android/platformdefines.h (hunk 7 above)
Comment (gaochun) [inline os/android/platformdefines.h]: @kalyankondapally It seems you comment at an obsolete patch, this code is not exit in the latest patch.

**Comment by gaochun** [comment]
Reviewer could not see the latest update, close this PR and try with a new one. @kalyankondapally

**Pull request closed without merging**